tutorials/appengine-ruby-rails-activejob-pubsub.md

```diff
@@ -0,0 +1,272 @@
+---
+title: Ruby on Rails Background Processing on Google App Engine with ActiveJob and Google Cloud Pub/Sub
+description: Learn how to run background jobs using Ruby on Rails' ActiveJob.
+author: chingor13
+tags: App Engine, Ruby, Ruby on Rails, ActiveJob, PubSub
+date_published: 2017-06-08
+---
+This tutorial shows how to create and configure a [Ruby on Rails](http://rubyonrails.org/) application to run
+background processing jobs on Google App Engine Flexible Environment using
+[ActiveJob](http://guides.rubyonrails.org/active_job_basics.html) and
+[Google Cloud Pub/Sub](https://cloud.google.com/pubsub/).
+
+## Objectives
+
+* Create a background processing job
+* Deploy your application to Google App Engine Flexible Environment
+* Verify background jobs are running
+
+## Before you begin
+
+You'll need the following:
+
+* A Google Cloud Console project. You can use an existing project or click the button to create a new project
+* [Ruby 2.4.0+ installed](https://www.ruby-lang.org/en/documentation/installation/)
+* A Rails 4.2+ application. Follow the
+  [official "Getting Started with Rails" guide](http://guides.rubyonrails.org/getting_started.html) to get started.
+* [Google Cloud SDK installed](https://cloud.google.com/sdk/downloads)
+
+## Costs
+
+This tutorial uses billable components of Cloud Platform including:
+
+* Google App Engine Flexible Environment
+* Google Cloud Pub/Sub
+
+Use the [pricing calculator](https://cloud.google.com/products/calculator/)
+to generate a cost estimate based on your projected usage. New Cloud Platform users might be eligible for a
+[free trial](https://cloud.google.com/free-trial).
+
+## Creating your background job
+
+Starting with version 4.2, Ruby on Rails includes an abstraction layer around background job processing called
+[ActiveJob](http://guides.rubyonrails.org/active_job_basics.html). This abstraction allows you to write the queuing
+and execution logic independently of queue and job runner implementations.
+
+Ruby on Rails provides command-line tools for generating templated skeletons for things such as database migrations,
+controllers, and even background jobs.
+
+You will create a job named `HelloJob` that will accept a `name` argument and "Hello #{name}" to standard output.
```

Comment:
Also, below the code prints "Hello, #{name} from ActiveJob!"

Update the 2 strings to match 😄